addons/account/tests/test_account_move_reconcile.py

```diff
@@ -5261,3 +5261,78 @@ def reconcile_move(move, transaction_amount, balance=None, date='2023-09-30', cu
             self.assertEqual(payment2.state, 'paid')
             reconcile_move(payment1.move_id, 12, lines_filter=lambda l: l.account_id.account_type not in ('asset_receivable', 'liability_payable'))
             self.assertEqual(payment1.state, 'paid')
+
+    def test_links_between_move_and_payment(self):
+        """ Test the field 'invoice_ids' of an account.payment (Many2many account.move 'matched_payment_ids') and ensure :
+        - It is always reset to None when the payment is reset to draft, even if the payment is confirmed again
+        - It is always filled with the right move(s) when we reconcile the payment with a move
+        """
+
+        def create_move_payment(move, payment_amount, with_outstanding_account=False):
+            payment = self.env['account.payment.register'].with_context(
+                active_model='account.move',
+                active_ids=move.ids
+            ).create({
+                'amount': payment_amount,
+                'payment_method_line_id':
+                    self.company_data['default_journal_bank'].inbound_payment_method_line_ids.filtered_domain([
+                        ('payment_account_id', '!=' if with_outstanding_account else "=", False),
+                    ])[0].id,
+            })._create_payments()
+            self.assertEqual(payment.state, 'in_process')
+            return payment
+
+        # No link when payment is reset to draft and confirmed again
+        invoice_outstanding_1 = self.init_invoice(move_type='out_invoice', amounts=[300], post=True)
```

Comment:
i'm wondering if we should not split the different test in multiple method and just put the helper as a normal function